src/corelibs/U2Gui/src/ObjectViewModel.h

```diff
@@ -246,6 +264,13 @@ protected slots:
     bool closing;
     QList<GObjectViewObjectHandler *> objectHandlers;
     OptionsPanel *optionsPanel;
+    QList<GObjectViewActionsProvider *> actionsProviders;
+};
+
+class U2GUI_EXPORT GObjectViewActionsProvider {
```

Comment:
Why do you need a separate class here?

Reply:
Separate from the GObjectViewWindowContext?

I think the reason is the same why GObjectViewObjectHandler separated form  GObjectViewWindowContext   - to separate a clean functionality instead of passing a bloated GObjectViewWindowContext . 

The GObjectViewActionsProvider should not be necessary an instance of the GObjectViewWindowContext.